tensorflow_datasets/core/registered_test.py

```diff
@@ -239,3 +239,23 @@ class SkipRegisteredDataset(object):
     name = "skip_registered_dataset"
     self.assertEqual(name, SkipRegisteredDataset.name)
     self.assertNotIn(name, load.list_builders())
+
+  def test_skip_regitration(self):
+    """Test `skip_registration()`."""
+
+    with testing.MockFs() as fs:
+      fs.add_file("datasets/ds1/config/1.0.0/temp.txt")
+      fs.add_file("datasets/ds2/1.0.0/temp.txt")
+
+      self.assertEqual(load.find_builder("ds1/config:1.0.0", data_dir="datasets"),
+                       "datasets/ds1/config/1.0.0")
+      self.assertEqual(load.find_builder("ds2:1.0.0", data_dir="datasets"),
+                       "datasets/ds2/1.0.0")
+      self.assertEqual(load.find_builder("ds1", data_dir="datasets"),
+                       "datasets/ds1/config/1.0.0")
+      self.assertEqual(load.find_builder("ds2", data_dir="datasets"),
+                       "datasets/ds2/1.0.0")
```

Comment:
We should add more test, specially when detecting failure mode. For instance
* when the ds name, config name / version do not exists (e.g. typo in the name)
* When a config is passed, but the dataset do not have one (`ds/config:1.2.0` but dataset is `ds:1.0.0`)
* When `ds:1.0.0` is passed for a dataset with config `ds/config:1.0.0`
* When no version is passed but multiple version exists (config and non-config version)